DoSomething/gambit-conversations
# Refactors loading Rivescript and parsing yes/no [pr]

#### What's this PR do?

This PR refactors the `menu` Rivescript topic as `ask_yes_no`, and renames the `confirmedTopic` and `declinedTopic` macros as `saidYes` and `saidNo`, and makes the following changes as a result:

* When loading Rivescript from gambit-campaigns on app start, we no longer create Rivescript topics that include the `menu` topic for each topic we want to ask if user wants to join (e.g. `askSignup` broadcast, MENU command)

    * Instead of writing the fetched Rivescript to file, loads it via the rivescript-js `stream` function - removes `brain/contentApi.rive` from `.gitignore`

* Adds a `helpers.request.parseAskYesNoResponse` that gets the bot reply for a message within the `ask_yes_no` topic, and calls this from the catchAll macro middleware to determine if user said yes or no if the last outbound message was a yes/no question

   * Updates the inbound message macro with `saidYes` or `saidNo` to salvage broadcast stats, as the `rivescript-reply-get` middleware is now setting the macro as `catchAll` now that our campaign topics no longer include the `menu` topic.


#### How should this be reviewed?
* Verify responses to askSignup/askContinue templates work as expected (we're still parsing yes/no correctly)
* Verify Rivescript from gambit-campaigns is loaded correctly upon app start (e.g. "who is this", "hi", "mirror" are all triggers defined in Contentful)

#### Any background context you want to provide?

* I've changed the `askSubscriptionStatus` topic to `ask_subscription_status` as the Rivescript parser throws an error if a topic name contains a capital letter. This error doesn't break anything as is, but ideally we should aim to update our rivescript-js to v2 and expect a Promise from the `brain.loadDirectory` call. The trouble there is that the bot will fail to load 

* This work is anticipation of supporting an `askChangeTopicBroadcast`, where we don't want to require a topic to require triggers (to include the `menu` topic) in order to parse yes/no responses to the broadcast

#### Relevant tickets
Starts on https://www.pivotaltracker.com/story/show/159413638, blocker for `askChangeTopicBroadcast` in https://www.pivotaltracker.com/story/show/157369418 

#### Checklist
